src/utils/theme.utils.ts

```diff
@@ -7,7 +7,7 @@ const themes = {
   dark: { name: 'dark', class: 'dark-mode' },
 }
 
-export const isDarkMode = () => themes.dark.name === LocalStorageService.get(LOCALSTORAGE_THEME_KEY)
+export const isDarkMode = () => themes.dark.name === LocalStorageService.get(LOCALSTORAGE_THEME_KEY) || (LocalStorageService.get(LOCALSTORAGE_THEME_KEY) == null && window.matchMedia('(prefers-color-scheme: dark)').matches)
```

Comment:
I think writing it this way is a bit more explicit. It's more code for sure, but that wacky boolean is gonna cause someone to actually have to stop and think about what's going on. Making it read a bit better will save the next dev the headache of having to figure it out.

```suggestion
export const isDarkMode = () => {
  // If they've set a theme, use that theme
  const localStorageTheme = LocalStorageService.get(LOCALSTORAGE_THEME_KEY)
  if (localStorageTheme) return themes.dark.name === localStorageTheme

  // If they have a system theme, use that
  const osDarkTheme = window.matchMedia('(prefers-color-scheme: dark)').matches
  if (osDarkTheme) return osDarkTheme

  // If no theme was detected return light theme
  return false
}
```

Reply:
that's fair but the variables seem redundant and over explicit IMO but im not going to fuss over it if you think otherwise

Reply:
What's redundant about this?

Reply:
i just dont like the 1 time use variables but im not really that invested in the one liner